fabric-convention-tags-v2/src/client/java/net/fabricmc/fabric/impl/tag/convention/ConventionLogWarningsClient.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.tag.convention;
+
+import java.util.List;
+
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minecraft.client.resource.language.I18n;
+import net.minecraft.item.Item;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKeys;
+import net.minecraft.registry.tag.TagKey;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
+import net.fabricmc.loader.api.FabricLoader;
+
+public class ConventionLogWarningsClient implements ClientModInitializer {
+	public static final Logger LOGGER = LoggerFactory.getLogger(ConventionLogWarningsClient.class);
+
+	/**
+	 * A config option mainly for developers.
+	 * Logs out modded item tags that do not have translations when running on integrated server.
+	 * Defaults to SILENCED.
+	 */
+	private static final String LOG_UNTRANSLATED_WARNING_MODE = System.getProperty("fabric-tag-conventions-v2.missingTagTranslationWarning", LOG_WARNING_MODES.SILENCED.name());
+	private enum LOG_WARNING_MODES {
```

Comment:
Why not CamelCase? :D

Reply:
You mean like `LogWarningModes`? I thought enums in Java tend to be done with uppercase underscores

Reply:
Oh the enum contents would be uppercase but enum itself is camel case. Oops. The same field in the legacy tag warning class is also upper case.